tracer/src/Datadog.Trace.SourceGenerators/InstrumentationDefinitions/InstrumentationDefinitionsGenerator.cs

```diff
@@ -0,0 +1,771 @@
+                        default:
+                            hasMisconfiguredInput = true;
+                            break;
+                    }
+
+                    if (hasMisconfiguredInput)
+                    {
+                        break;
+                    }
+                }
+
+                if (hasMisconfiguredInput)
+                {
+                    continue;
+                }
+
+                if (assemblyNames is null or { Length: 0 } && assemblyName is null)
+                {
+                    hasMisconfiguredInput = true;
+                    diagnostics ??= new List<Diagnostic>();
+                    diagnostics.Add(
+                        MissingRequiredPropertyDiagnostic.Create(
+                            attributeData.ApplicationSyntaxReference?.GetSyntax(),
+                            Constants.InstrumentAttributeProperties.AssemblyName,
+                            Constants.InstrumentAttributeProperties.AssemblyNames));
+                }
+
+                if (typeNames is null or { Length: 0 } && typeName is null)
+                {
+                    hasMisconfiguredInput = true;
+                    diagnostics ??= new List<Diagnostic>();
+                    diagnostics.Add(
+                        MissingRequiredPropertyDiagnostic.Create(
+                            attributeData.ApplicationSyntaxReference?.GetSyntax(),
+                            Constants.InstrumentAttributeProperties.TypeName,
+                            Constants.InstrumentAttributeProperties.TypeNames));
+                }
+
+                if (integrationName is null)
+                {
+                    hasMisconfiguredInput = true;
+                    diagnostics ??= new List<Diagnostic>();
+                    diagnostics.Add(
+                        MissingRequiredPropertyDiagnostic.Create(
+                            attributeData.ApplicationSyntaxReference?.GetSyntax(),
+                            Constants.InstrumentAttributeProperties.IntegrationName));
+                }
+
+                if (methodName is null)
+                {
+                    hasMisconfiguredInput = true;
+                    diagnostics ??= new List<Diagnostic>();
+                    diagnostics.Add(
+                        MissingRequiredPropertyDiagnostic.Create(
+                            attributeData.ApplicationSyntaxReference?.GetSyntax(),
+                            Constants.InstrumentAttributeProperties.MethodName));
+                }
+
+                if (returnTypeName is null)
+                {
+                    hasMisconfiguredInput = true;
+                    diagnostics ??= new List<Diagnostic>();
+                    diagnostics.Add(
+                        MissingRequiredPropertyDiagnostic.Create(
+                            attributeData.ApplicationSyntaxReference?.GetSyntax(),
+                            Constants.InstrumentAttributeProperties.MethodName));
```

Comment:
```suggestion
                            Constants.InstrumentAttributeProperties.ReturnTypeName));
```

Reply:
Fixed in https://github.com/DataDog/dd-trace-dotnet/pull/2288/commits/bb06fe0e6040cc2707f475d33e03a3e300719ddf, thanks 👍